### ISSUE TITLE
Improve test repository.yml

### ISSUE THREAD

**Issue by bdestombe** [NHFLO/data#13]
Test if all fields are present
Test if versions are sorted

```python
def compare_semver(version1: str, version2: str) -> int:
    """
    Compare two semantic version strings according to semver.org rules.

    Parameters
    ----------
    version1 : str
        First semantic version string to compare (e.g., "1.2.3" or "1.2.3-alpha.1")
    version2 : str
        Second semantic version string to compare (e.g., "1.2.3" or "1.2.3-beta.1")

    Returns
    -------
    int
        1 if version1 is newer than version2
        -1 if version1 is older than version2
        0 if they are equal

    Raises
    ------
    ValueError
        If either version string doesn't follow semantic versioning format

    Notes
    -----
    Follows semantic versioning 2.0.0 specification from semver.org:
    - Versions are compared by major.minor.patch numbers first
    - Pre-release versions have lower precedence than normal versions
    - Pre-release versions are compared alphanumerically by identifiers
    - Numeric identifiers are compared numerically
    - Alphabetic identifiers are compared alphabetically

    Examples
    --------
    >>> compare_semver("2.1.0", "2.0.0")
    1
    >>> compare_semver("1.0.0", "2.0.0")
    -1
    >>> compare_semver("1.0.0", "1.0.0")
    0
    >>> compare_semver("1.0.0-alpha", "1.0.0")
    -1
    >>> compare_semver("1.0.0-alpha.1", "1.0.0-alpha.2")
    -1
    >>> compare_semver("1.0.0-alpha.beta", "1.0.0-alpha.1")
    1

    See Also
    --------
    https://semver.org/ : Official semantic versioning specification
    """
    def parse_version(version: str) -> tuple:
        # Split version into parts and prerelease if exists
        if '-' in version:
            version_part, prerelease = version.split('-', 1)
        else:
            version_part, prerelease = version, ''
            
        # Parse version numbers
        try:
            major, minor, patch = map(int, version_part.split('.'))
        except ValueError:
            raise ValueError(f"Invalid version format: {version}")
            
        return (major, minor, patch, prerelease)
    
    try:
        v1_major, v1_minor, v1_patch, v1_pre = parse_version(version1)
        v2_major, v2_minor, v2_patch, v2_pre = parse_version(version2)
        
        # Compare major.minor.patch
        v1_tuple = (v1_major, v1_minor, v1_patch)
        v2_tuple = (v2_major, v2_minor, v2_patch)
        
        if v1_tuple > v2_tuple:
            return 1
        if v1_tuple < v2_tuple:
            return -1
            
        # If major.minor.patch are equal, check pre-release
        # No pre-release is greater than any pre-release version
        if not v1_pre and v2_pre:
            return 1
        if v1_pre and not v2_pre:
            return -1
        if v1_pre == v2_pre:
            return 0
            
        # Compare pre-release versions
        # Split by dots and compare each identifier
        v1_pre_parts = v1_pre.split('.')
        v2_pre_parts = v2_pre.split('.')
        
        for p1, p2 in zip(v1_pre_parts, v2_pre_parts):
            # Try to convert to integers if possible
            try:
                n1 = int(p1)
                n2 = int(p2)
                if n1 != n2:
                    return 1 if n1 > n2 else -1
            except ValueError:
                # Compare as strings if not both numbers
                if p1 != p2:
                    return 1 if p1 > p2 else -1
        
        # If one prerelease version has more parts, it's newer
        return 1 if len(v1_pre_parts) > len(v2_pre_parts) else -1 if len(v1_pre_parts) < len(v2_pre_parts) else 0
            
    except Exception as e:
        raise ValueError(f"Invalid version format: {str(e)}")
```

**Comment by bdestombe** [comment]
```python
import unittest
from typing import List, Tuple

class TestSemverCompare(unittest.TestCase):
    def assert_version_order(self, versions: List[str]):
        """Helper method to verify that a list of versions is in ascending order"""
        for i in range(len(versions) - 1):
            result = compare_semver(versions[i], versions[i + 1])
            self.assertEqual(
                result, -1,
                f"Expected {versions[i]} to be older than {versions[i + 1]}"
            )

    def test_basic_comparison(self):
        """Test basic version number comparisons"""
        test_cases = [
            ("1.0.0", "2.0.0", -1),
            ("2.0.0", "1.0.0", 1),
            ("2.1.0", "2.0.0", 1),
            ("2.0.1", "2.0.0", 1),
            ("1.0.0", "1.0.0", 0),
        ]
        
        for v1, v2, expected in test_cases:
            with self.subTest(v1=v1, v2=v2):
                self.assertEqual(compare_semver(v1, v2), expected)

    def test_prerelease_comparison(self):
        """Test pre-release version comparisons"""
        test_cases = [
            ("1.0.0-alpha", "1.0.0", -1),
            ("1.0.0", "1.0.0-alpha", 1),
            ("1.0.0-alpha", "1.0.0-beta", -1),
            ("1.0.0-beta", "1.0.0-alpha", 1),
            ("1.0.0-alpha.1", "1.0.0-alpha.2", -1),
            ("1.0.0-alpha.2", "1.0.0-alpha.1", 1),
            ("1.0.0-alpha.1", "1.0.0-alpha.1", 0),
        ]
        
        for v1, v2, expected in test_cases:
            with self.subTest(v1=v1, v2=v2):
                self.assertEqual(compare_semver(v1, v2), expected)

    def test_mixed_numeric_string_prerelease(self):
        """Test mixed numeric and string pre-release identifiers"""
        test_cases = [
            ("1.0.0-alpha.1", "1.0.0-alpha.beta", -1),
            ("1.0.0-alpha.beta", "1.0.0-beta.2", -1),
            ("1.0.0-beta.2", "1.0.0-beta.11", -1),
            ("1.0.0-beta.11", "1.0.0-rc.1", -1),
        ]
        
        for v1, v2, expected in test_cases:
            with self.subTest(v1=v1, v2=v2):
                self.assertEqual(compare_semver(v1, v2), expected)

    def test_prerelease_length_comparison(self):
        """Test comparison of pre-release versions with different numbers of identifiers"""
        test_cases = [
            ("1.0.0-alpha", "1.0.0-alpha.1", -1),
            ("1.0.0-alpha.1", "1.0.0-alpha.1.1", -1),
            ("1.0.0-alpha.1.1", "1.0.0-alpha.1", 1),
        ]
        
        for v1, v2, expected in test_cases:
            with self.subTest(v1=v1, v2=v2):
                self.assertEqual(compare_semver(v1, v2), expected)

    def test_invalid_versions(self):
        """Test that invalid version formats raise ValueError"""
        invalid_versions = [
            "1.0",
            "1.0.0.0",
            "1",
            "a.b.c",
            "1.0.0-",
            "-1.0.0",
            "1.0.0-alpha..1",
            "",
            "not_a_version"
        ]
        
        valid_version = "1.0.0"
        for invalid_version in invalid_versions:
            with self.subTest(invalid_version=invalid_version):
                with self.assertRaises(ValueError):
                    compare_semver(invalid_version, valid_version)
                with self.assertRaises(ValueError):
                    compare_semver(valid_version, invalid_version)

    def test_version_ordering(self):
        """Test that a sequence of versions is correctly ordered"""
        ordered_versions = [
            "1.0.0-alpha",
            "1.0.0-alpha.1",
            "1.0.0-alpha.beta",
            "1.0.0-beta",
            "1.0.0-beta.2",
            "1.0.0-beta.11",
            "1.0.0-rc.1",
            "1.0.0",
            "1.0.1",
            "1.1.0",
            "2.0.0",
        ]
        self.assert_version_order(ordered_versions)

if __name__ == '__main__':
    unittest.main()
```

**Comment by bdestombe** [comment]
```python
"""Tests for YAML schema validation and linting.

This module validates both schema compliance using yamale and style rules using yamllint.
"""

import pytest
import yamale
from yamale.validators import DefaultValidators, Validator
from yamllint import config as lint_config, linter

class PathsMatchVersion(Validator):
    """Ensure all paths end with version_nhflo."""
    
    def _is_valid(self, value):
        version_nhflo = self.value_context.get('version_nhflo')
        if not version_nhflo:
            return False
            
        return (value['local'].endswith(version_nhflo) and
                value['nhflo_server'].endswith(version_nhflo) and
                value['mockup'].endswith(version_nhflo))

def test_yaml_file():
    """Test both schema validation and linting rules."""
    # Read YAML content
    with open('your_data.yaml', 'r') as f:
        yaml_content = f.read()
    
    # Schema validation
    validators = DefaultValidators.copy()
    validators['paths_match_version'] = PathsMatchVersion
    
    schema = yamale.make_schema('nhflo_schema.yaml', validators=validators)
    data = yamale.make_data(content=yaml_content)
    yamale.validate(schema, data)
    
    # Lint validation
    lint_conf = lint_config.YamlLintConfig("""
        extends: default
        rules:
            document-start: disable
            line-length: disable
            empty-lines:
                max: 1
                max-start: 1
                max-end: 1
            indentation:
                spaces: 2
                indent-sequences: true
            braces:
                min-spaces-inside: 0
                max-spaces-inside: 0
            brackets:
                min-spaces-inside: 0
                max-spaces-inside: 0
            comments:
                min-spaces-from-content: 2
    """)
    
    problems = list(linter.run(yaml_content, lint_conf))
    
    # Format lint problems into readable message if any exist
    if problems:
        error_msg = "\nLinting problems found:\n"
        for problem in problems:
            error_msg += f"Line {problem.line}: {problem.message}\n"
        pytest.fail(error_msg)

if __name__ == "__main__":
    pytest.main([__file__, "-v"])
```

**Comment by bdestombe** [comment]
solved with #16
